File: meilleurdev/2017-session1/building/building.py
#!/usr/local/bin/python3
# 
# #*******
#* Read input from STDIN
#* Use echo or print to output your result, use the /n constant at the end of each result line.
#* Use:
#*      local_print (variable ); 
#* to display simple variables in a dedicated area.
#* ***/
import sys
from collections import *
import heapq 
import itertools
import copy

sys.setrecursionlimit(15000)

debug = True

def dprint(*args):
    if debug:
	    print(args)

lines = []
for line in sys.stdin:
	lines.append(line.rstrip('\n'))


bruteforce?

File: meilleurdev/2017-session1/building/building.cpp
/*******
 * Read input from cin
 * Use cout to output your result.
 * Use:
 *  localPrint( string mystring)
 * to display variable in a dedicated area.
 * ***/
#include <iostream>
#include <limits>
#include <sstream>
#include <string>
#include <iostream>
#include <algorithm>
#include <vector>
#include <map>
#include <set>
#include "exercise.hpp"

using namespace std;

ContestExerciseImpl::ContestExerciseImpl() : Exercise() {}

void ContestExerciseImpl::main() {
	int L;
	cin >> L;
	int N;
	cin >> N;
	map<int, int> F;
	int res = 0;
	vector<int> V;
	for (int i = 0; i < N; i++) {
		int t;
		cin >> t;
		V.push_back(t);
	}

	for (int i = 0; i < N; i++) {
		if (V[i] == L) {
			res++;
			continue;
		}
		for (int j = i + 1; j < N; j++) {
			if (V[i] + V[j] == L) {
				res++;
				V[i] = 0;
				V[j] = 0;
				break;
			}
		}
	}


	cout << res << endl;



    // BRUTEFORCE?

}